simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/ConfirmationFragment.kt

```diff
@@ -0,0 +1,175 @@
+    }
+
+    // Go to next screen
+    binding.nextBtn.setOnClickListener {
+      logger.debug("User created card")
+      returnResult()
+    }
+
+    // Go to previous screen
+    binding.prevBtn.setOnClickListener {
+
+      if (ContextCompat.checkSelfPermission(activity!!, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        != PackageManager.PERMISSION_GRANTED
+      ) {
+        logger.debug("Requesting storage permission")
+        ActivityCompat.requestPermissions(
+          activity!!,
+          arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+          storageRequestCode
+        )
+      } else {
+        createDigitalCard()
+      }
+    }
+  }
+
+  /**
+   * Create library card from ImageView
+   */
+  private fun createDigitalCard() {
+    val card = binding.libraryCard
+    card.isDrawingCacheEnabled = true
+    val bitmap = card.drawingCache
+    val f: File
+    try {
+      if (Environment.getExternalStorageState() == Environment.MEDIA_MOUNTED) {
+        val file = File(Environment.getExternalStorageDirectory(), Environment.DIRECTORY_PICTURES)
+        if (!file.exists()) {
+          file.mkdirs()
+        }
+        f = File(file.absolutePath + File.separator + "library-card" + ".png")
+        val outStream = FileOutputStream(f)
+        bitmap.compress(Bitmap.CompressFormat.PNG, 10, outStream)
+        outStream.close()
+        addCardToGallery(f)
+        Toast.makeText(activity!!, getString(R.string.card_saved), Toast.LENGTH_SHORT).show()
+      }
+    } catch (e: Exception) {
+      e.printStackTrace()
+    }
+  }
+
+  /**
+   * Adds library card image to device photo gallery
+   */
+  private fun addCardToGallery(card: File) {
+    val mediaScanIntent = Intent(Intent.ACTION_MEDIA_SCANNER_SCAN_FILE)
+    val contentUri: Uri = Uri.fromFile(card)
+    mediaScanIntent.data = contentUri
+    activity!!.sendBroadcast(mediaScanIntent)
+  }
+
+  /**
+   * Listen for result from location permission request, this method is a callback provided by
+   * Android for the requestPermissions() method
+   *
+   * @param requestCode - String user defined request code to identify the request
+   * @param permissions - String Array of permissions requested
+   * @param grantResults - Integer array of what the user has granted/denied
+   */
+  override fun onRequestPermissionsResult(
+    requestCode: Int,
+    permissions: Array<String>,
+    grantResults: IntArray
+  ) {
+    when (requestCode) {
+      storageRequestCode -> {
+        // If request is cancelled, the result arrays are empty.
+        if ((grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+          logger.debug("Storage permission granted")
+          createDigitalCard()
+        } else {
+          logger.debug("Storage permission NOT granted")
+        }
+        return
+      }
+    }
+  }
+
+  /**
+   * Returns result to caller with card details
+   */
+  private fun returnResult() {
+    val data = Intent()
```

Comment:
Would be a little more idiomatic to use "apply" here:

```kotlin
val data = Intent().apply {
  putExtra("type", type)
  // ...
  putExtra("message", message)
}
```

Reply:
prettier